src/swarm/neo/client/mixins/AllNodesRequestCore.d

```diff
@@ -0,0 +1,648 @@
+        private bool connect ( )
+        {
+            return allNodesRequestConnector(this.conn);
+        }
+    }
+}
+
+/*******************************************************************************
+
+    Struct template encapsulating the core logic for initialising a single
+    request-on-conn of an all-nodes request. Initialisation is deinfed as:
+        * Sending a message containing all information required by the node to
+          start handling the request.
+        * Waiting for the status code (a single ubyte) sent back by the node.
+        * Validating the status code.
+
+    The behaviours at each stage are defined by a set of policy instances, the
+    types of which are defined by template arguments.
+
+    This struct is suitable for passing to AllNodesRequest as the Initialiser
+    policy.
+
+    Note that this initialiser requires a field named all_nodes, of type
+    AllNodesRequestSharedWorkingData, to exist in the request's shared working
+    data. The fields of this struct are used by the initialiser and should not
+    be touched by other code.
+
+    Params:
+        Request = type of the request being initialised. The request's shared
+            working data is expected to contain a field of type
+            AllNodesRequestSharedWorkingData called all_nodes
+        FillPayload = type of policy instance to be called to add any required
+            fields to the initial message payload sent to the node
+        HandleStatusCode = type of policy instance to be called to validate the
+            status code received from the node. Should return true to continue
+            handling the request or false to abort
+
+*******************************************************************************/
+
+public struct AllNodesRequestInitialiser ( Request, FillPayload,
+    HandleStatusCode )
+{
+    import swarm.neo.client.RequestOnConn;
+    import swarm.neo.connection.RequestOnConnBase;
+
+    /// Connection event dispatcher.
+    public RequestOnConn.EventDispatcherAllNodes conn;
+
+    /// Request context.
+    public Request.Context* context;
+
+    /// Policy object for adding fields to the initial message payload.
+    public FillPayload fill_payload;
+
+    /// Policy object for handling the status code received from the node.
+    public HandleStatusCode handle_status_code;
+
+    /***************************************************************************
+
+        Performs the logic of request initialisation. Called by
+        AllNodesRequestHandler.
+
+        Returns:
+            true if initialisation succeeded for this request-on-conn, false to
+            abort handling of the request on this connection
+
+    ***************************************************************************/
+
+    public bool initialise ( )
+    {
+        // establishConnection() should guarantee we're already connected
+        assert(this.conn.waitForReconnect() == 0);
```

Comment:
I almost said "there shouldn't be any sideeffects inside `assert`, since it would be removed when compiling with `-release`", but this is essentially a way to say `assert(this.conn.isConnected())`. Maybe we should make that clearer (not in this PR).